x/gov/keeper/msg_server_test.go

```diff
@@ -187,6 +192,10 @@ func (suite *KeeperTestSuite) TestVoteReq() {
 				res, err := suite.msgSrvr.SubmitProposal(suite.ctx, msg)
 				suite.Require().NoError(err)
 				suite.Require().NotNil(res.ProposalId)
+
+				// reset proposal number, we ignore the dynamic deposit system in these tests
+				suite.govKeeper.SetActiveProposalsNumber(suite.ctx, 0)
```

Comment:
So I was wondering why this is needed because the test cases are supposed to reset the state between them, so any increase of the deposit in one case should not affect the other case. Unfortunately it appears this is not the case and this damned test suite sounds buggy af. I will investigate more to understand what's happenning there.

Reply:
The investigation demonstrates that `KeeperTestSuite` does not reset its state between cases, which is very bad because test cases step on each others. I decided to fix that here #107, as a result, this PR also removes the calls to `SetActiveProposalNumber` you had to add.